### DISCUSSION TITLE
Receive data form otle

### DISCUSSION THREAD

**Discussion by Roc-00** [apache/skywalking#7814] [Q&A] [answered]
Hi, I am a student participate in ‘summer of code’,this is my first time.This is my first contact with skywalking.I am trying to use skywalking to collect the data from open telemetry. And I had read this:
![image](https://user-images.githubusercontent.com/84842773/134861564-efb25403-3432-4878-9e35-80551593f023.png)
What about the data of trace?
And I want to know are there any documents about starting sky walking in IDEA？

**Comment by wu-sheng** [comment, marked as answer]
Hi, as a `Summer of Code` student, you should have a mentor right? Who is he/she?

> What about the data of trace?

We just accept data from Zipkin, which is not analyzable, we only receive, save, and provide query.

**Reply by Roc-00** [reply]
I am from eventMesh.My mentor is easonchen.This is my project:https://summer.iscas.ac.cn/#/org/prodetail/210650513
I'm not sure I understand correctly.Is it because the trace data format of otle is different from the data format received by skywalking? Can the data exported by otle be transformed into data that skywalking can analyze, so as to realize the visualization of data?

**Reply by wu-sheng** [reply]
In your proposal, there is nothing related to SkyWalking. Could you be more clear what are you asking for?

SkyWalking only analysis its own trace format, due to what we introduced in STAM paper, https://wu-sheng.github.io/STAM/.

But no matter what we supported or not, this seems out of scope.

**Reply by Roc-00** [reply]
I mean, otle can have different exporters to visualize data. 
Among the tasks I have completed, metrics uses Prometheus for visualization and trace uses Zipkin and log. My mentor hopes I can adapt skywalking and make a skywalking plug-in implementation.

**Reply by wu-sheng** [reply]
If you want to expose otel trace to Zipkin format, SkyWalking server could accept and visualize, but it wouldn't analysis. Dapper format traces(Zipkin and Jaeger) are not suitable for analysis. We have announced this multiple times publicly.

**Reply by Roc-00** [reply]
Because I am a newcomer in the field of distributed tracking, I came into contact with APM only after participating in the SOC activity. Therefore, sky walking not only visualizes the data, but also makes special analysis on the data. That's what I didn't know before. I will ask my tutor if I only need visualization. Thank you very much for your answer, which let me understand new knowledge

**Reply by wu-sheng** [reply]
Zipkin is a tracing system, SkyWalking is an APM. Those are different.

**Reply by wu-sheng** [reply]
> sky walking not only visualizes the data

This is not a correct conclusion. Visualizing trace is working. The missing part is analysis, meaning, metrics/topology can't be generated.

**Reply by Roc-00** [reply]
You mean that the data can only be visualized after analysis?

**Reply by wu-sheng** [reply]
No, APM includes trace visualization. `be visualized after analysis` is not the correct call.

**Reply by Roc-00** [reply]
、


> > sky walking not only visualizes the data
> 
> This is not a correct conclusion. Visualizing trace is working. The missing part is analysis, meaning, metrics/topology can't be generated.

This means that if skywalking receives the trace data of otle, it can only realize visualization, not analysis ,not metrics/topology,right?

**Reply by wu-sheng** [reply]
Yes.
